src/client/pre.js

```diff
@@ -1,13 +1,15 @@
 const copyButton = document.createElement("template");
-copyButton.innerHTML = '<button title="Copy code" class="observablehq-pre-copy"><svg width="16" height="16" viewBox="0 0 16 16" fill="none" stroke="currentColor" stroke-width="2"><path d="M2 6C2 5.44772 2.44772 5 3 5H10C10.5523 5 11 5.44772 11 6V13C11 13.5523 10.5523 14 10 14H3C2.44772 14 2 13.5523 2 13V6Z M4 2.00004L12 2.00001C13.1046 2 14 2.89544 14 4.00001V12"></path></svg></button>'; // prettier-ignore
+
+copyButton.innerHTML = '<button title="Copy code" class="observablehq-pre-copy"><span class="observablehq-data-copied">Copied</span><svg width="16" height="16" viewBox="0 0 16 16" fill="none" stroke="currentColor" stroke-width="2"><path d="M2 6C2 5.44772 2.44772 5 3 5H10C10.5523 5 11 5.44772 11 6V13C11 13.5523 10.5523 14 10 14H3C2.44772 14 2 13.5523 2 13V6Z M4 2.00004L12 2.00001C13.1046 2 14 2.89544 14 4.00001V12"></path></svg></button>'; // prettier-ignore
 
 enableCopyButtons();
+const copiedNotice = document.querySelector(".observablehq-data-copied");
 
 export function enableCopyButtons() {
   for (const pre of document.querySelectorAll("pre:not([data-copy=none])")) {
     const parent = pre.parentNode;
-    if (parent.classList.contains("observablehq-pre-container")) continue;
     const div = parent.insertBefore(document.createElement("div"), pre);
+    if (parent.classList.contains("observablehq-pre-container")) continue;
```

Comment:
why do you have to move this?

Reply:
to be honest I'm not sure if this works. but I put it back because I thought I deleted it by mistake.

should it be removed?

Reply:
If you have doubts, we should definitely double check that it works, but my question is simply, why are you suggesting to switch these two lines?

Reply:
Actually I don't suggest changing it. that line was already there before. 

I thought it solved a problem that I can't see now, so I chose to add it again. I hope I understood correctly.

Reply:
If you check the full diff you'll see that it is now below the line that reads
`const div = parent.insertBefore(document.createElement("div"), pre);`
when it was previously above it.

Reply:
Oh sorry, I completely misunderstood. I don't know where my mind was, now I looked at the original code and saw the mistake. :(

Reply:
i fix now